__tests__/app.test.js

```diff
@@ -0,0 +1,71 @@
+const request = require("supertest");
+const db = require("../db/connection");
+const app = require("../app");
+const {
+  topicData,
+  userData,
+  articleData,
+  commentData,
+} = require("../db/data/test-data/index");
+const seed = require("../db/seeds/seed");
+const jsonEndpoints = require("../endpoints.json");
+
+beforeEach(() => seed({ topicData, userData, articleData, commentData }));
+afterAll(() => db.end());
+
+describe("GET/api/topics", () => {
+  it("GET:200 status with an array of all topic objects", () => {
+    return request(app)
+      .get("/api/topics")
+      .expect(200)
+      .then(({ body }) => {
+        const { topics } = body;
+        expect(topics).toHaveLength(3);
+        topics.forEach((topic) => {
+          expect(topic).toMatchObject({
+            description: expect.any(String),
+            slug: expect.any(String),
+          });
+        });
+      });
+  });
+});
+
+describe("GET/api", () => {
+  it("provide a description of all other endpoints available", () => {
+    return request(app)
+      .get("/api")
+      .expect(200)
+      .then(({ body }) => {
+        const keys = Object.keys(body);
+        const endpoints = Object.keys(jsonEndpoints);
+        const results = endpoints.some((end) => keys.includes(end));
+        expect(results).toBe(true);
+        for (let i = 0; i < keys.length; i++) {
+          expect(keys[i]).toBe(endpoints[i]);
+        }
+      });
+  });
+});
+
+describe("GET/api/articles/:article_id", () => {
+  it("get an article by its id", () => {
+    return request(app)
+      .get("/api/articles/1")
+      .expect(200)
+      .then(({ body }) => {
+        const sample = {
+          author: "butter_bridge",
+          title: "Living in the shadow of a great man",
+          article_id: 1,
+          body: "I find this existence challenging",
+          topic: "mitch",
+          created_at: "2020-07-09T20:11:00.000Z",
```

Comment:
For this one, I would suggest changing the structure of the test a little to use expect.ObjectContaining as you can then use expect.any('string') for the date and still check the specific key values.